docs/docs/sourcing-from-etsy.md

```diff
@@ -0,0 +1,113 @@
+
+Note that Etsy expects developers to test the API using real data. That means your shop and listings will be live even as you're working on this site. If you're already an active Etsy seller, this shouldn't change your usual workflow much. If you're starting a new shop, be mindful of [Etsy's testing policies](https://www.etsy.com/developers/documentation/getting_started/testing#section_etsy_api_testing_policies) as you test your work.
+
+## Using the Etsy source plugin
+
+You can use the `gatsby-source-etsy` plugin to pull featured listings and images from your Etsy shop into your Gatsby site. To install it, run:
+
+```bash
+npm install gatsby-source-etsy
+```
+
+### Configuring the plugin
+
+You will then need to add the plugin, your API key, and your shop ID to your `gatsby-config.js` file. That said, finding your shop ID can be confusing. While your shop does have an ID _number_, you don't actually need this to make requests against the Etsy API. You can use your shop's _name_ as a unique identifier if you don't know your shop ID.
+
+```js:title=gatsby-config.js
+plugins: [
+  {
+    resolve: "gatsby-source-etsy",
+    options: {
+      // highlight-start
+      apiKey: "your api key here",
+      shopId: "your shop id or shop name here",
+      // highlight-end
+      language: "en", // optional
+    },
+  },
+],
+```
+
+### Testing your queries
+
+Once you have everything set up, run `gatsby develop` to start your site locally. At that point, you should be able to query for data related to your featured listings in [GraphiQL](/docs/running-queries-with-graphiql/). As an example, the following query gets the total number of featured listings in the shop as well as the price, title, and description of each item.
+
+```graphql
+allFeaturedEtsyListing {
+  totalCount
+    nodes {
+      price
+      title
+      description
+    }
+  }
+```
+
+Make sure you have _featured_ items! To edit your store, go to the Shop Manager and select your Etsy shop under "Sales Channels". You should see an option to edit or enable featured items.
+
+## Displaying Etsy listings
+
+You can pull your Etsy listing data into a page or component by composing a GraphQL query and specifying the data you would like to display. For example, consider the following query, which gets the title, price, URL, and an image of each featured listing.
+
+```graphql
+allFeaturedEtsyListing {
+  nodes {
+    title
+    price
+    url
+    description
+    childEtsyListingImage {
+      childFile {
+        childImageSharp {
+          fixed(width: 400, height: 400) {
+            ...GatsbyImageSharpFixed
+          }
+        }
+      }
+    }
+  }
+}
+```
+
+Using this query on a page or in a component will allow you to display your listings. The following example loops through the featured items and displays them within a component.
+
+```js
+{
+  data.allFeaturedEtsyListing.nodes.map(item => (
```

Comment:
I believe this will throw a warning in React because there isn't a unique key for each item. We may not care since it's a guide, but perhaps an index as key makes sense to include?